### PR TITLE
Sroberge/dev

### PR DESCRIPTION
**Pull request by vim-sroberge**

### CODE REVIEW

**Comment by vim-sroberge** [comment]
Outlines and material refactor
Settings refactor
Docs regen

**Comment by vim-sroberge** [comment]
Outline and post-process implementation
Material refactor and folder
Object gfx effects refactor by introducing objectAttribute
viewer Settings refactor to be a simple object
loadMore for rooms
render on demand
replaced lambdas with signals
more exports for doc

File: src/vim-loader/vim.ts [hunk 1]
```diff
@@ -15,18 +15,17 @@ import { Object } from './object'
 export class Vim {
   source: string | undefined
   readonly document: IDocument
-  index: number = -1
-  settings: VimSettings
+  settings: VimConfig
```
Comment (mavimaec) [inline src/vim-loader/vim.ts:18]: Minor - "settings" field could be renamed to "config" to keep the nomenclature consistent with VimConfig

**Review by mavimaec — APPROVED** [review]
Awesome work @vim-sroberge !

**Pull request merged**